Tennis-scheduler/src/main/java/com/tennis/tennisscheduler/controllers/PersonController.java

```diff
@@ -59,13 +64,25 @@ public ResponseEntity<PersonDto> getPersonById(@PathVariable long id){
         return new ResponseEntity<>(personDtoMapper.fromPersonToPersonDto(person),HttpStatus.OK);
     }
     @PutMapping("/{id}")
+    @PreAuthorize("hasAnyRole('TENNIS_PLAYER','ADMIN')")
     public ResponseEntity<PersonDto> updatePerson(@PathVariable long id,@RequestBody PersonDto personDto){
+
         Person personExisting = personService.findById(id);
-        if (personExisting == null) {
-            return new ResponseEntity<>( HttpStatus.BAD_REQUEST);
-        }
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        Person user = (Person) authentication.getPrincipal();
 
-        Person person = personService.updatePerson(id, personDtoMapper.fromPersonDtoToPerson(personDto));
-        return new ResponseEntity<>(personDtoMapper.fromPersonToPersonDto(person),HttpStatus.OK);
+        if(id==user.getId()) {
```

Comment:
The code in if and else if is basically the same. The only thing different is that in the first one you are using user.getId() which should be the same as id sent in the request if the TENNIS_PLAYER is trying to update their own profile. No need for duplication.
What you can do is to create separate if for checking the role and the id, like you had above for checking if person exists in the database (which you should leave as was). If the id is different return 401, if not just let it do the same thing as ADMIN.

Reply:
thanks for the suggestion :)